### PR TITLE
Fix incorrect parsing of Int values

### PR DESCRIPTION
**Pull request by rbanerjee**
There was an error in tril when trying to parse a large hex value (such as  0xFFFFFFFF00000000). This corrects that, and makes the parser test more robust.

### CODE REVIEW

File: fvtest/tril/test/ParserTest.cpp [hunk 1]
```diff
@@ -26,26 +26,33 @@
...
+    ASSERT_STREQ(GetParam().second, trees->getName());
     ASSERT_NULL(trees->getArgs());
     ASSERT_NULL(trees->getChildren());
     ASSERT_NULL(trees->next);
 }
 
+// The generated tests from this instantiation will have the following names:
+// ParserTest/SingleNodeWithName.VerifyNode/0 for TestParamWithExpectedResultString("(nodeName)", "nodename") ..
+// With the convention being:
+// InstantiationName/FooTest.DoesBlah/n for val_n
```
Comment (Leonardo2718) [inline fvtest/tril/test/ParserTest.cpp]: Not sure this comment is needed.

Comment (Leonardo2718) [inline fvtest/tril/test/ParserTest.cpp]: Not sure this comment is needed, but I guess it doesn't hurt to have around.

File: fvtest/tril/test/ParserTest.cpp [hunk 2]
```diff
@@ -100,24 +107,16 @@ TEST(ParserTest, NodeWithChildAndSibling) {
     ASSERT_NULL(sibling->next);
 }
 
-TEST(ParserTest, SingleNodeWithIntArg) {
-    auto trees = parseString("(nodeName 3)");
+// aliases for the underlying C++ types used to store the different AST types.
+using Integer_t = uint64_t;
+using FloatingPoint_t = double;
```
Comment (Leonardo2718) [inline fvtest/tril/test/ParserTest.cpp]: Could you use `ASTValue::Integer_t` and `ASTValue::FloatingPoint_t` instead? This will guarantee that the types are the same as the ones used by Tril.

File: fvtest/tril/test/ParserTest.cpp [hunk 3]
```diff
@@ -128,29 +127,35 @@ TEST(ParserTest, SingleNodeWithHexArg) {
...
     ASSERT_EQ(ASTValue::Integer, arg->getValue()->getType());
-    ASSERT_EQ(0xabc123, arg->getValue()->getInteger());
+    ASSERT_EQ(GetParam().second, arg->getValue()->getInteger());
     ASSERT_NULL(arg->next);
 }
 
-TEST(ParserTest, SingleNodeWithNegativeHexArg) {
-    auto trees = parseString("(nodeName -0x249DEF)");
+INSTANTIATE_TEST_CASE_P(
+    ParserTest,
```
Comment (Leonardo2718) [inline fvtest/tril/test/ParserTest.cpp]: For consistency with the other test, I would rather this be `ParserTestSingleNodeWithIntArgAsDecValue`.

File: fvtest/tril/test/ParserTest.cpp [hunk 4]
```diff
@@ -161,10 +166,18 @@ TEST(ParserTest, SingleNodeWithFloatArg) {
...
+    ASSERT_EQ(GetParam().second, arg->getValue()->getFloatingPoint());
     ASSERT_NULL(arg->next);
 }
 
+INSTANTIATE_TEST_CASE_P(
+    ParserTest,
+    SingleNodeWithFloatArg,
+    testing::Values(
+        TestParamWithExpectedResultFloat("(nodeName 3.00)", 3.00),
+        TestParamWithExpectedResultFloat("(nodeName -3.00)", -3.00)
```
Comment (Leonardo2718) [inline fvtest/tril/test/ParserTest.cpp]: Might be worth having a few (at least one) edge cases here.

File: fvtest/tril/tril/tril.l [hunk 5]
```diff
@@ -43,7 +43,7 @@
             };
 
 [-]?0x[0-9a-fA-F]+      {
-                yylval.integer = strtol(yytext, NULL, 16);
+                yylval.integer = strtoul(yytext, NULL, 16);
```
Comment (Leonardo2718) [inline fvtest/tril/tril/tril.l]: Good catch!

File: fvtest/tril/test/ParserTest.cpp (hunk 1 above)
Comment (rbanerjee) [reply fvtest/tril/test/ParserTest.cpp]: I'm okay either way, don't think it matters that much. @mgaudet, thoughts?

Comment (mgaudet) [reply fvtest/tril/test/ParserTest.cpp]: I'd say pull the comment -- just because it's documenting the expected behaviour of GoogleTest, and so isn't adding a huge amount value for experienced readers (though, I understand how it would help people new to the project; we just can't optimize the code for them, or else the code would be so thick with explanatory comment that it would be difficult to see what's going on)

**Comment by mgaudet** [comment]
Speaking of more edge cases: 

```
diff --git a/fvtest/tril/test/ParserTest.cpp b/fvtest/tril/test/ParserTest.cpp
index 7075d0e2..21113b17 100644
--- a/fvtest/tril/test/ParserTest.cpp
+++ b/fvtest/tril/test/ParserTest.cpp
@@ -138,7 +138,8 @@ INSTANTIATE_TEST_CASE_P(
         TestParamWithExpectedResultInt("(nodeName 0)", 0),
         TestParamWithExpectedResultInt("(nodeName 3)", 3),
         TestParamWithExpectedResultInt("(nodeName -3)", -3),
-        TestParamWithExpectedResultInt("(nodeName 9223372036854775807)", 9223372036854775807U)
+        TestParamWithExpectedResultInt("(nodeName 9223372036854775807)", 9223372036854775807U),
+        TestParamWithExpectedResultInt("(nodeName 18446744073709551615)", 18446744073709551615ULL)
 ));
 
```

produces

```

[----------] 5 tests from ParserTest/SingleNodeWithIntArg
../fvtest/tril/test/ParserTest.cpp:130: Failure
      Expected: GetParam().second
      Which is: 18446744073709551615
To be equal to: arg->getValue()->getInteger()
      Which is: 9223372036854775807
[  FAILED  ] ParserTest/SingleNodeWithIntArg.VerifyNode/4, where GetParam() = (0x107d491ae pointing to "(nodeName 18446744073709551615)", 18446744073709551615) (0 ms)
```

**Comment by rbanerjee** [comment]
Addressed the comments except for the one asking for more tests that test the parsing for floating point numbers. I will that as a separate pull request  

Also modified the previous commit messages to reflect that this now fixes parsing issues with Integers(represented as *both hex and decimal) that are longer than 'long'.
